From 9c964d03a9c857a2d3b67be2619e0cc3d3eba8e5 Mon Sep 17 00:00:00 2001
From: "Matthieu Baerts (NGI0)" <matttbe@kernel.org>
Date: Wed, 20 Nov 2024 12:20:07 +0100
Subject: [PATCH 1/4] include: define __bswap_constant_{16,32} if needed

Some libc like musl don't support them.

Let's simply define them if they are not defined.

From GNU C Library, bits/byteswap.h, under LGPL.

Signed-off-by: Matthieu Baerts (NGI0) <matttbe@kernel.org>
---
 include/mptcpd/private/sockaddr.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/mptcpd/private/sockaddr.h b/include/mptcpd/private/sockaddr.h
index 1b8de34..5d74ef8 100644
--- a/include/mptcpd/private/sockaddr.h
+++ b/include/mptcpd/private/sockaddr.h
@@ -26,6 +26,16 @@
  */
 ///@{
 #if __BYTE_ORDER == __LITTLE_ENDIAN
+/* These 2 helpers come from GNU C Library, under LGPL */
+# ifndef __bswap_constant_16
+#  define __bswap_constant_16(x)				\
+    ((__uint16_t) ((((x) >> 8) & 0xff) | (((x) & 0xff) << 8)))
+# endif
+# ifndef __bswap_constant_32
+#  define __bswap_constant_32(x)				\
+    ((((x) & 0xff000000u) >> 24) | (((x) & 0x00ff0000u) >> 8)	\
+     | (((x) & 0x0000ff00u) << 8) | (((x) & 0x000000ffu) << 24))
+# endif
 # define MPTCPD_CONSTANT_HTONS(hs) __bswap_constant_16(hs)
 # define MPTCPD_CONSTANT_HTONL(hl) __bswap_constant_32(hl)
 #else
-- 
2.45.2

